### ISSUE TITLE
API calls hanging with 1.3.0-rc4

### ISSUE THREAD

**Issue by KaiRo-at** [NebulousLabs/Sia#2122]
I installed 1.3.0-rc4 on a Minebox dev VM tofay and sometimes immediately, sometimes after some time of running, any connection to the siad API (including siac of course) is hanging and not returning anything.

Sometimes I see a message like this in the console:

```
A lock was held for too long, id '31'. Call stack:
        File: '/home/luke/go/src/github.com/NebulousLabs/Sia/modules/renter/repair.go:476'
        File: '/home/luke/go/src/github.com/NebulousLabs/Sia/modules/renter/repair.go:305'
```

Also, after a few times of killing siad when hanging and restarting, I now get this message as well:

```
goroutine 102 [running]:
runtime/debug.Stack(0x4457b7, 0x0, 0xc42b797c28)
/usr/local/go/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Critical(0xc42b797d80, 0x2, 0x2)
/home/luke/go/src/github.com/NebulousLabs/Sia/build/critical.go:16 +0xa7
github.com/NebulousLabs/Sia/persist.(*Logger).Critical(0xc4216f7a40, 0xc42b797d80, 0x2, 0x2)
/home/luke/go/src/github.com/NebulousLabs/Sia/persist/log.go:37 +0xe0
github.com/NebulousLabs/Sia/modules/renter.(*Renter).managedDownloadIteration(0xc4216e7080, 0xc42b797f20)
/home/luke/go/src/github.com/NebulousLabs/Sia/modules/renter/download.go:359 +0x5ce
github.com/NebulousLabs/Sia/modules/renter.(*Renter).threadedDownloadLoop(0xc4216e7080)
/home/luke/go/src/github.com/NebulousLabs/Sia/modules/renter/download.go:622 +0x37e
created by github.com/NebulousLabs/Sia/modules/renter.newRenter
/home/luke/go/src/github.com/NebulousLabs/Sia/modules/renter/renter.go:219 +0x3ae
Critical error: ERROR: the renter is idle , but tracking active pieces: 2
Please submit a bug report here: https://github.com/NebulousLabs/Sia/issues
```

This may be directly connected or due to the restarts, I can't really tell ;-)

**Comment by KaiRo-at** [comment]
Apparently the lock message comes with different line numbers, e.g. this one:

A lock was held for too long, id '224'. Call stack:
File: '/home/luke/go/src/github.com/NebulousLabs/Sia/modules/renter/repair.go:199'
File: '/home/luke/go/src/github.com/NebulousLabs/Sia/modules/renter/repair.go:566'

Also, it looks like "only" the /consensus and any /renter/... endpoints are blocked, others like wallet or hostdb ones still work.

**Comment by lukechampine** [comment]
the deadlock and active piece bugs should be fixed in v1.3.0. Please reopen if they are still reproducible.
